src/Shared/UnitTests/ObjectModelHelpers.cs

```diff
@@ -442,16 +442,14 @@ internal static void AssertArrayContentsMatch(object[] expected, object[] actual
             if (expected == null)
             {
                 Assert.Null(actual); // "Expected a null array"
-            }
-            else
-            {
-                Assert.NotNull(actual); // "Result should be non-null."
+                return;
             }
 
+            Assert.NotNull(actual); // "Result should be non-null."
             Assert.Equal(expected.Length, actual.Length); // "Expected array length of <" + expected.Length + "> but was <" + actual.Length + ">.");
 
             // Now that we've verified they're both non-null and of the same length, compare each item in the array.
-            for (int i = 0; i < expected.Length; i++)
+            for (int i = expected.Length - 1; i >= 0; i--)
```

Comment:
The other changes to this function are good, but why change the loop to iterate from end -> beginning?